src/platform/w3dengine/client/w3dtreebuffer.cpp

```diff
@@ -804,18 +804,18 @@ void W3DTreeBuffer::Load_Trees_In_Index_And_Vertex_Buffers(
 
                                         unsigned int color;
 
-                                        if (color_array) {
+                                        if (color_array != nullptr) {
                                             color = color_array[j];
                                         } else {
                                             color = 0xFFFFFFFF;
                                         }
 
-                                        vertexes->diffuse = Do_Lighting(&loc, lighting, &emissive, color, 1.0f);
+                                        vertexes->diffuse = Do_Lighting(&loc, lighting, &emissive, color, 1.0);
```

Comment:
This change makes no sense, 1.0f is correct here.

Reply:
If I remember correctly there was a mention of Original Game using float and double as the same type, aka compiler turns both to 4 bytes?

If that is true, shouldn't all floats in game be 4 byte floats and never 8 byte doubles?

Reply:
there are doubles, this is a case of me doing the whole function from scratch and arriving at the same code